src/parser/jobs/nin/Ninki.js

```diff
@@ -0,0 +1,102 @@
+import React, {Fragment} from 'react'
+//import {Icon, Message} from 'semantic-ui-react'
+
+//import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+//import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+// Constants
+const MAX_NINKI = 100
+const NINKI_SPENDERS = [
+	ACTIONS.HELLFROG_MEDIUM.id,
+	ACTIONS.BHAVACAKRA.id,
+	ACTIONS.TEN_CHI_JIN.id,
+]
+
+export default class Ninki extends Module {
+	static handle = 'ninki'
+	static dependencies = [
+		'suggestions',
+	]
+	
+	_ninki = 0
+	_wastedNinki = 0
+	_wasteBySource = {
+		mug: 0,
+		auto: 0,
+	}
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('cast', {by: 'player'}, this._onCast)
+		//this.addHook('death', {to: 'player'}, this._onDeath)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onCast(event) {
+		const abilityId = event.ability.guid
+		
+		if (abilityId === ACTIONS.MUG.id) {
+			this._wasteBySource.mug += this._addNinki(30)
```

Comment:
Can I just get the 30 for Mug and the 6 for AAs moved into a const obj up top? Should make it easier to make a quick change if/when required, and can simplify the code by just checking if the abilityId is in the object.

Stretch goal to also move the ninki spenders into the same system, but that's less of an issue.

Reply:
Can do. Another case of magic numbers, I suppose.